src/main/java/com/bootcamp/demo/validation/UserValidator.java

```diff
@@ -0,0 +1,109 @@
+package com.bootcamp.demo.validation;
+
+import com.bootcamp.demo.model.User;
+
+public class UserValidator {
+
+    public final static String NAME_PATTERN = "[A-Z]+[a-z]+[A-Z]*[a-z]*";
+    public final static String EMAIL_PATTERN = "[a-zA-Z0-9._-]+@[a-zA-Z]+[.][a-zA-Z]+";
+    public final static String CONTAINING_UPPERCASE_PATTERN = ".*[A-Z]+.*";
+    public final static String CONTAINING_LOWERCASE_PATTERN = ".*[a-z]+.*";
+    public final static String CONTAINING_DIGIT_PATTERN = ".*[0-9]+.*";
+    public final static String CONTAINING_SPECIAL_CHARACTER_PATTERN = ".*[!#$%^&*~_=+./<>-]+.*";
+    public final static String PHONE_NUMBER_PATTERN = "[+]40[0-9]{9}";
+
+    public String validateFirstName(String firstName) {
+        String errors = "";
+        if (firstName == null) {
+            errors += "First name cannot be empty!\n";
+        }
+        else if (!firstName.matches(NAME_PATTERN)) {
+            errors += "First name must start with an uppercase letter and have at least another letter!\n";
+        }
+        return errors;
+    }
+
```

Comment:
You have a lot of repetitive code, which is a sign that you could extract the functionality into some methods, and use them in all places needed. For example, you had to add the "\n" to each error message by hand (if by mistake you forget it into one place, the entire error message will not be properly generated).

In your case, if you would define 3 helper methods, one that validates string (if it's empty), one that validates a pattern and one that validates the length, then you could use them for all the validate methods.

Checking if a String is null is not enough, you should always check if it's null or empty (there are some helper methods in the StringUtils class that you could use).

For example:
```
private String validateString(String field, String errorMessage) {
        String validationResult = "";
        if (StringUtils.isEmpty(field)) {
            validationResult = errorMessage + NEW_LINE;
        }
        return validationResult;
    }

    private String validatePattern(String field, String pattern, String errorMessage) {
        String validationResult = "";
        if (isNotEmpty(field) && !field.matches(pattern)) {
            validationResult = errorMessage + NEW_LINE;
        }
        return validationResult;
    }
```
then your validate methods would become something simple and easy to read:
```
    public String validateFirstName(String firstName) {
        return String.join("",
                validateString(firstName, "First name cannot be empty"),
                validatePattern(firstName, NAME_PATTERN, "First name must start with an uppercase letter and have at least another letter")
        );
    }

    public String validateLastName(String lastName) {
        return join("",
                validateString(lastName, "Last name cannot be empty"),
                validatePattern(lastName, NAME_PATTERN, "Last name must start with an uppercase letter and have at least another letter!")
        );
    }

    public String validatePassword(String password) {
        return join("",
                validateString(password, "Password cannot be empty"),
                validateLength(password, PASSWORD_LENGTH, "Password isn't long enough!"),
                validatePattern(password, CONTAINING_UPPERCASE_PATTERN, "Password must contain an uppercase letter!"),
                validatePattern(password, CONTAINING_LOWERCASE_PATTERN, "Password must contain a lowercase letter!"),
                validatePattern(password, CONTAINING_DIGIT_PATTERN, "Password must contain a digit!"),
                validatePattern(password, CONTAINING_SPECIAL_CHARACTER_PATTERN, "Password must contain a special character(!#$%^&*~_=+./<>-)")
        );
    }
```